modules/scripts/startup-script/main.tf

```diff
@@ -45,13 +45,23 @@ locals {
   # Final content output to the user
   stdlib = join("", local.stdlib_list)
 
-  runners_map = { for runner in var.runners :
+  runners_map = { for runner in local.runners :
     basename(runner["destination"])
     => {
       content = lookup(runner, "content", null)
       source  = lookup(runner, "source", null)
     }
   }
+
+  ansible_installer = {
+    type        = "shell"
+    source      = "${path.module}/examples/install_ansible.sh"
+    destination = "install_ansible.sh"
+  }
+
+  ansible_local_runners     = [for r in var.runners : r if r.type == "ansible-local"]
+  prepend_ansible_installer = length(local.ansible_local_runners) > 0 && var.prepend_ansible_installer
+  runners                   = local.prepend_ansible_installer ? concat([local.ansible_installer], var.runners) : var.runners
```

Comment:
super-nit: I typically find it more elegant to define variables before they are used (call it a reminder of my mid 90's C programming). So I would have defined these lines at the top... ;)

Reply:
Addressed nit while I was addressing a build failure.